xilem/examples/virtual_cats.rs

```diff
@@ -53,17 +51,46 @@ impl VirtualCats {
             .get_mut(index)
             .expect("VirtualScroll bounds set correctly.");
         let img = match &item.image {
-            ImageState::NotRequested => {
-                // We can't store the items to be scheduled in `self`, because the parent view isn't re-run when this function is executed
-                // This is for the reasonable reason that that could create a loop, if this weren't conditional.
-                // However, we probably should have an escape hatch for when this is conditional.
-                self.tx.as_ref().unwrap().blocking_send(item.code).unwrap();
-                item.image = ImageState::Pending;
-                None
-            }
             ImageState::Pending => None,
             ImageState::Available(image) => Some(image),
         };
+        let task = if img.is_none() {
+            // Capturing the code is valid here, because this will never change for this view.
+            let code = item.code;
+            // If the cat is not loaded yet, we create a task to load it.
+            // N.B. If we wanted to batch requests, the lifecycle unfortunately gets a
+            // lot more complicated currently.
+            // Note that if this item scrolls offscreen, the ongoing load will be cancelled.
+            Some(xilem::view::task_raw(
+                move |proxy| {
+                    async move {
+                        let url = format!("https://http.cat/{code}");
+                        // Add an artificial delay to show how variable sizes work
+                        tokio::time::sleep(Duration::from_millis(300)).await;
+                        let result = image_from_url(&url).await;
+                        match result {
+                            // We choose not to handle the case where the event loop has ended
+                            Ok(image) => drop(proxy.message(image)),
+                            // TODO: Report in the frontend
+                            Err(err) => {
+                                tracing::warn!(
+                                    "Loading image for HTTP status code {code} from {url} failed: {err:?}"
+                                );
+                            }
+                        }
+                    }
+                },
+                move |state: &mut Self, image| {
+                    if let Some(status) = state.statuses.iter_mut().find(|it| it.code == code) {
+                        status.image = ImageState::Available(image);
+                    } else {
+                        // TODO: Error handling?
```

Comment:
What happens right now in the event of a failure?
Would it make sense to load a stock failure image?

Reply:
The spinner continues forever, as far as I remember. I don't think that loading a stock failure image is doable, because what if fail to load that!